### ISSUE TITLE
java.net.ProtocolException: Too many tunnel connections attempted: 21

### ISSUE THREAD

**Issue by dankarization** [instagram4j/instagram4j#601]
## Issue Template Requirements

- instagram4j 2.0.4 version
---

### Pre-Checklist

I made sure...
- [+ ] I read the README.md
- [+ ] I looked over the examples and wiki
- [+ ] my issue hasn't been asked or solved before

---

### Describe the bug

Without changing anything in my own code, last weeks I started to get **java.net.ProtocolException: Too many tunnel connections attempted: 21** error on almost all actions creating, deleting or interacting with content. This is clearly not a proxy issue because nothing like this was showing earlier on the same proxies and I get this on several localhost IP's and all proxy providers I was testing.

### How to reproduce

You can try to upload a new photo/ like post/ delete your own posts.
```
### Logs
Here are some logs with IG json payload
[2021-07-17.log](https://github.com/instagram4j/instagram4j/files/6834239/2021-07-17.log)

**Comment by jvogit** [comment]
https://github.com/square/retrofit/issues/1561#issuecomment-178696473

Could be related to this. Maybe Instagram is blocking by infinitely redirecting?

**Comment by dankarization** [comment]
May be. But if this is true- what can cause this ban? I consulted with a few people- all of them said this could be only library problem, not my code over it.

**Comment by jvogit** [comment]
> May be. But if this is true- what can cause this ban? I consulted with a few people- all of them said this could be only library problem, not my code over it.

Yes this could be. Might be missing new Headers or prelogin/postlogin requests. Perhaps the device may be the problem.

All these things require looking into the app to see the flows and mimicking them exactly, which unfortunately takes too much time for me 😅

so there might be new things added. unfortunately it's the price to pay for undocumented internal API as it needs constant updating or monitoring to see if there are new changes.
